File: forJavaLabs/Lab3/FilmStudio/src/manager/StudioManager.java
package manager;

import java.util.LinkedList;
import java.util.List;

import FilmStudio.Worker;
import model.enums.FilmGenre;

public class StudioManager {
	public  List <Worker> availableWorkers = new LinkedList <Worker>();
	
	public void print ()
	{
		
			for (Worker worker : availableWorkers) {
				System.out.println(worker);
			}
		}
	public StudioManager () {}
	
	public void addWorker (Worker addWorker) {
		availableWorkers.add(addWorker);
	}
	public List<Worker> getWorkerGenres() {
				return availableWorkers;
			}
			public void setAvailableWorkers(List<Worker> availableWorkers) {
				this.availableWorkers = availableWorkers;
			}
	
			public  List<Worker> findWorkersByGenres (String workerToFind) 
			{
				List<Worker> workerList = new LinkedList<Worker>();
				
				for (Worker worker : availableWorkers) {
					if(workerToFind.equals (worker.getGenre().name())) {
						workerList.add(worker);
					}
				}
				return workerList;
			}
			
			
			public List<Worker> sortBySalaryPerHour () {
				  availableWorkers.sort((Worker o1, Worker o2)->o1.getSalaryPerHour()-o2.getSalaryPerHour());
				  return availableWorkers;
			} 
			public StudioManager (LinkedList <Worker> availableWorkers) {
				this.availableWorkers = availableWorkers; 
			}
		}

File: forPythonLabs/Lab2/StudioManager.py
class StudioManager():
    available_workers = []

    def init(self):
        pass

    def add_worker(self, worker):

        self.available_workers += worker

    def print_(self):
        for worker in self.available_workers:
            print(worker.to_string())

    def print_list(self):
        for worker in self.available_workers:
            print(worker.to_string())

    def sort_by_salary_per_hour(self):
        self.available_workers.sort(key=lambda x: x.salary_per_hour, reverse=False)

    def find_workers_by_genres(self, film_genre):
        worker_list = []
        for worker in self.available_workers:
            if worker.film_genre == film_genre:
                worker_list.append(worker)
            return worker_list

        pass
